src/GraphQL.Conventions/Adapters/Resolvers/EventStreamResolver.cs

```diff
@@ -1,30 +1,26 @@
-﻿using GraphQL.Conventions.Types.Descriptors;
-using GraphQL.Resolvers;
-using GraphQL.Subscription;
+﻿using GraphQL.Resolvers;
 using System;
 using System.Threading.Tasks;
 
 namespace GraphQL.Conventions.Adapters.Resolvers
 {
-    public class EventStreamResolver : FieldResolver, IEventStreamResolver
+    public class EventStreamResolver : ISourceStreamResolver
     {
-        public EventStreamResolver(GraphFieldInfo fieldInfo) : base(fieldInfo)
+        private readonly IFieldResolver _fieldResolver;
+
+        public EventStreamResolver(IFieldResolver fieldResolver)
         {
+            _fieldResolver = fieldResolver;
         }
 
-        public IObservable<object> Subscribe(IResolveEventStreamContext context)
+        public async ValueTask<IObservable<object>> ResolveAsync(IResolveFieldContext context)
         {
-            var result = base.Resolve(context);
+            var result = await _fieldResolver.ResolveAsync(context);
             if (result is Task<object>)
             {
                 result = (result as Task<object>).Result;
             }
             return (IObservable<object>)result;
         }
-
-        public override object Resolve(IResolveFieldContext context)
-        {
-            return context.Source;
-        }

```

Comment:
This is critical for subscription support.

Reply:
Thanks a lot @Shane32